parser/mongodatastore.go

```diff
@@ -28,26 +30,30 @@ type collectionStore struct {
 
 //MongoDatastore is a datastore which stores bro data in MongoDB
 type MongoDatastore struct {
-	dbMap      map[string]map[string]collectionStore
-	bufferSize int
-	session    *mgo.Session
-	logger     *log.Logger
-	waitgroup  *sync.WaitGroup
-	mutex1     *sync.Mutex
-	mutex2     *sync.Mutex
+	dbMap       map[string]map[string]collectionStore
+	existingDBs []string
+	metaDB      *database.MetaDBHandle
+	bufferSize  int
+	session     *mgo.Session
+	logger      *log.Logger
+	waitgroup   *sync.WaitGroup
+	mutex1      *sync.Mutex
+	mutex2      *sync.Mutex
 }
 
 //NewMongoDatastore creates a datastore which stores bro data in MongoDB
-func NewMongoDatastore(session *mgo.Session,
+func NewMongoDatastore(session *mgo.Session, metaDB *database.MetaDBHandle,
 	bufferSize int, logger *log.Logger) *MongoDatastore {
 	return &MongoDatastore{
-		dbMap:      make(map[string]map[string]collectionStore),
-		bufferSize: bufferSize,
-		session:    session,
-		logger:     logger,
-		waitgroup:  new(sync.WaitGroup),
-		mutex1:     new(sync.Mutex), //mutex1 syncs the first level of map access
-		mutex2:     new(sync.Mutex), //mutex2 syncs the second level of map access
+		dbMap:       make(map[string]map[string]collectionStore),
+		existingDBs: metaDB.GetDatabases(),
```

Comment:
existingDBs holds the databases that existed before we called import. If an entry is destined for an already existing db, we exit with an error